Dockerfile.rosbag

```
@@ -0,0 +1,48 @@
+FROM ros:galactic-ros1-bridge
+ENV DEBIAN_FRONTEND noninteractive
+
+# ROS
+ENV ROS1_DISTRO noetic
+ENV ROS2_DISTRO galactic
+
+# ROS2
+RUN set -x && \
+  sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list' && \
+  apt-get update -y -qq && \
+  : "install ROS1 and ROS2" && \
+  apt-get install -y -qq \
+    ros-${ROS2_DISTRO}-image-transport \
+    ros-${ROS2_DISTRO}-cv-bridge \
+    ros-${ROS2_DISTRO}-image-tools \
+    ros-foxy-rosbag2-bag-v2-plugins \
+    ros-foxy-ros2bag \
+    ros-foxy-rosbag2-transport \
+    ros-foxy-rosbag2-converter-default-plugins \
+    ros-foxy-rosbag2-storage-default-plugins \
+    python3-pip \
+    python3-colcon-common-extensions && \
+  pip3 install -U \
+    argcomplete && \
+  : "remove cache" && \
+  apt-get autoremove -y -qq && \
+  rm -rf /var/lib/apt/lists/*
+
+WORKDIR /ros2_ws
+
+ARG NUM_THREADS=8
+RUN set -x && \
+  : "build ROS2 packages" && \
+  bash -c "source /opt/ros/${ROS_DISTRO}/setup.bash; \
+  colcon build --parallel-workers ${NUM_THREADS} --cmake-args \
+    -DBUILD_WITH_MARCH_NATIVE=ON \
+    -DUSE_PANGOLIN_VIEWER=OFF \
+    -DUSE_SOCKET_PUBLISHER=ON \
+    -DUSE_STACK_TRACE_LOGGER=ON \
+    -DBOW_FRAMEWORK=DBoW2"
+
+RUN set -x && \
+  sh -c "echo '#'\!'/bin/bash\nset -e\nsource /opt/ros/${ROS_DISTRO}/setup.bash\nsource /ros2_ws/install/setup.bash\nexec \"\$@\"' > /ros_entrypoint.sh" && \
+  chmod u+x /ros_entrypoint.sh
+
+ENTRYPOINT ["/ros_entrypoint.sh"]
```

The problem mentioned at https://github.com/OpenVSLAM-Community/openvslam_ros/pull/31/files#r666297717 is solved by the following change.

```suggestion
ENTRYPOINT []
```

---

I tried below in my rosbag:
```
FROM ros:galactic-ros1-bridge
ENV DEBIAN_FRONTEND noninteractive

# ROS2
RUN set -x && \
  apt-get update -y -qq && \
  : "install ROS1 and ROS2" && \
  apt-get install -y -qq \
    ros-${ROS2_DISTRO}--image-transport \
    ros-${ROS2_DISTRO}--cv-bridge \
    ros-${ROS2_DISTRO}--rosbag2-bag-v2-plugins \
    ros-${ROS2_DISTRO}--ros2bag \
    ros-${ROS2_DISTRO}--rosbag2-transport \
    ros-${ROS2_DISTRO}--rosbag2-storage-default-plugins \
    python3-pip \
    python3-colcon-common-extensions && \
  pip3 install -U \
    argcomplete && \
  : "remove cache" && \
  apt-get autoremove -y -qq && \
  rm -rf /var/lib/apt/lists/*

ENTRYPOINT []
CMD ["bash"]
```
**Build error:**
```
E: Unable to locate package ros-galactic--image-transport
E: Unable to locate package ros-galactic--cv-bridge
E: Unable to locate package ros-galactic--rosbag2-bag-v2-plugins
E: Unable to locate package ros-galactic--ros2bag
E: Unable to locate package ros-galactic--rosbag2-transport
E: Unable to locate package ros-galactic--rosbag2-storage-default-plugins
```
To solve above, i had to bring back below line:
`sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list' && \`

Now it is working.

---

> Build error:
> ```
> E: Unable to locate package ros-galactic--image-transport
> E: Unable to locate package ros-galactic--cv-bridge
> E: Unable to locate package ros-galactic--rosbag2-bag-v2-plugins
> E: Unable to locate package ros-galactic--ros2bag
> E: Unable to locate package ros-galactic--rosbag2-transport
> E: Unable to locate package ros-galactic--rosbag2-storage-default-plugins
> ```
> To solve above, i had to bring back below line:
> `sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list' && \`

I can build it without `sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list' && \` in my environment. What is the difference?

---

It looks like there is one more hyphen.

---

Could you please tell me which hyphen you are talking about? I am unable to figure it out.

---

@ymd-stella 
Deleted above line as suggested and rechecked, build is through and i am able to use ros2 bag play info command.
Sorry for the confusion.

Uploaded final Dockerfile.rosbag for review.